### DISCUSSION TITLE
Can not remove breakpoint when debugging

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-cdt-cloud/cdt-cloud#42] [Q&A]
I set a breakpoint in  a loop code when debugging I click to remove the breakpoint, but I find it will be triggered at the point in the next loop
```C
#include "Rectangle.h"
#include <windows.h>

void Rectangle::set_values(int x, int y)
{
  width = x;
  height = y;
}

void Rectangle::sleep(int millisecond)
{
  Sleep(millisecond);
}

int Rectangle::area()
{
  return width * height;
}
```

```C
#include "Rectangle.h"
#include <iostream>
using namespace std;

int main()
{
  Rectangle rect, rectb;
  rect.set_values(3, 4);
  rectb.set_values(5, 6);
  cout << "rect area: " << rect.area() << endl;
  cout << "rectb area: " << rectb.area() << endl;

  int i = 20;
  while (i > 10)
  {
    rect.sleep(1000);
    cout << i << endl;
  }
  
  return 0;
}
```
![2024-07-02_14-36-48](https://github.com/eclipse-cdt-cloud/cdt-cloud/assets/4537809/352e7ca1-4657-42e4-87b2-e6928d183f25)

**Comment by planger** [comment]
Hi,

I assume this happens with the [cdt-gdb-adapter](https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter), correct?
It might be helpful if you also post the GBD logs to further diagnose the issue.

@jonahgraham Do you have an idea to help out?

Thank you!

**Reply by wss29** [reply]
hi @planger 
   yes it is cdt-gdb-adapter, I download [cdt-cloud-blueprint](https://github.com/eclipse-cdt-cloud/cdt-cloud-blueprint) and build it run in  browser. the project is also from [here](https://github.com/eclipsesource/cdtcloud-alpha/tree/master/example/workspace) but add a sleep loop for test
[workspace.zip](https://github.com/user-attachments/files/16063364/workspace.zip)

**Reply by wss29** [reply]
Hi @planger 
this is a log I save, I am not sure whether is what you need
[a.log](https://github.com/user-attachments/files/16063773/a.log)

**Reply by jonahgraham** [reply]
The log was super helpful to identify the issue. The issue could be solved in two places, for the adapter side I raised https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/issues/330

However Theia should try to send canonical paths to the adapter as this issue may affect all adapters. VSCode has a similar problem a long time ago, see https://github.com/microsoft/vscode/issues/43959

**Reply by jonahgraham** [reply]
Reading through the history of https://github.com/microsoft/vscode/issues/43959 and all the related bugs makes me think that at a minimum the adapter should normalize the drive letter even if Theia also adds the normalization.
